rust/src/proofs/api.rs

```diff
@@ -402,6 +404,166 @@ pub unsafe extern "C" fn fil_seal_commit_phase2(
     })
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn fil_seal_commit_phase2_for_aggregation(
+    seal_commit_phase1_output_ptr: *const u8,
+    seal_commit_phase1_output_len: libc::size_t,
+    sector_id: u64,
+    prover_id: fil_32ByteArray,
+) -> *mut fil_SealAggregationCommitPhase2Response {
+    catch_panic_response(|| {
+        init_log();
+
+        info!("seal_commit_phase2_for_aggregation: start");
+
+        let mut response = fil_SealAggregationCommitPhase2Response::default();
+
+        let scp1o = serde_json::from_slice(from_raw_parts(
+            seal_commit_phase1_output_ptr,
+            seal_commit_phase1_output_len,
+        ))
+        .map_err(Into::into);
+
+        let result = scp1o.and_then(|o| {
+            seal_commit_phase2_for_aggregation(o, prover_id.inner, SectorId::from(sector_id))
+        });
+
+        match result {
+            Ok((output, inputs)) => {
+                response.status_code = FCPResponseStatus::FCPNoError;
+                response.proof_ptr = output.proof.as_ptr();
+                response.proof_len = output.proof.len();
+
+                // Given that we have a Vec<Vec<u8>>, this flattens it into a
+                // single u8 array, marking the length of each inner stride
+                // for reconstruction, as they may be uneven values
+                let mut input_lens = Vec::with_capacity(inputs.len());
+                let flat_inputs = inputs.iter().fold(Vec::new(), |mut acc, input| {
+                    let flat_inner_inputs =
+                        input.iter().fold(Vec::new(), |mut inner_acc, element| {
+                            inner_acc.extend(fr_into_bytes(&element));
+                            inner_acc
+                        });
+
+                    input_lens.push(flat_inner_inputs.len());
+                    acc.extend(flat_inner_inputs);
+                    acc
+                });
+
+                response.inputs.num_inputs = inputs.len();
+                response.inputs.input_lens_ptr = input_lens.as_ptr();
+                response.inputs.input_ptr = flat_inputs.as_ptr();
+
+                mem::forget(output.proof);
+                mem::forget(input_lens);
+                mem::forget(flat_inputs);
+            }
+            Err(err) => {
+                response.status_code = FCPResponseStatus::FCPUnclassifiedError;
+                response.error_msg = rust_str_to_c_str(format!("{:?}", err));
+            }
+        }
+
+        info!("seal_commit_phase2_for_aggregation: finish");
+
+        raw_ptr(response)
+    })
+}
+
+pub unsafe extern "C" fn fil_aggregate_seal_proofs(
```

Comment:
It seems like the Go method for this didn't get generated

Reply:
Ok, I've got most of the re-factor working locally, but am not sure how it affects your commit.  I will likely open a new draft PR on a different branch and we can resolve which is better later.